traits/trait_types.py

```diff
@@ -2774,6 +2774,178 @@ def items_event(self):
 AdaptMap = {"no": 0, "yes": 1, "default": 2}
 
 
+class Map(TraitType):
+    """ Checks that the value assigned to a trait attribute is a key of a
+        specified dictionary, and also assigns the dictionary value
+        corresponding to that key to a *shadow* attribute.
+
+        A trait attribute of type Map is called a *mapped* trait
+        attribute. In practice, this means that the resulting object actually
+        contains two attributes: one whose value is a key of the Map
+        dictionary, and the other whose value is the corresponding value of the
+        Map dictionary. The name of the shadow attribute is simply the base
+        attribute name with an underscore ('_') appended. Mapped trait
+        attributes can be used to allow a variety of user-friendly input values
+        to be mapped to a set of internal, program-friendly values.
+
+        Example
+        -------
+
+        The following example defines a ``Person`` class::
+
+            >>>class Person(HasTraits):
+            ...    married = Map({'yes': 1, 'no': 0 })
+            >>>
+            >>>bob = Person()
+            >>>print bob.married
+            yes
+            >>>print bob.married_
+            1
+
+        In this example, the default value of the ``married`` attribute of the
+        Person class is 'yes'. Because this attribute is defined using
+        PrefixList, instances of Person have another attribute,
+        ``married_``, whose default value is 1, the dictionary value
+        corresponding to the key 'yes'.
```

Comment:
Changed `PrefixList` to `PrefixMap`. But I don't feel that the default value is obvious, since we consider the first item in the map as the default but it isn't explicitly mentioned.